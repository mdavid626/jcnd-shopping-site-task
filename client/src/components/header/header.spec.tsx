import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ShoppingCartButton from '../shopping-cart-button/shopping-cart-button';
import Header from './header';

jest.mock('../shopping-cart-button/shopping-cart-button');

describe('header', () => {
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  it('should render', () => {
    (ShoppingCartButton as jest.Mock).mockReturnValue(
      <div>shopping-cart-button</div>
    );
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
