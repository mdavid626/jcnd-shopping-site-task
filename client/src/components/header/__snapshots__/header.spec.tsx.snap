// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <div
      class="Header-logoAndTitle"
    >
      <img
        alt="logo"
        class="Header-logo"
        src="logo.png"
      />
      <div
        class="Header-title"
      >
        Simple Shopping Site
      </div>
    </div>
    <div
      class="Header-avatarAndUser"
    >
      <img
        alt="avatar"
        class="Header-avatar"
        src="avatar.svg"
      />
      <div
        class="Header-user"
      >
        Max Pecu
      </div>
    </div>
    <div>
      shopping-cart-button
    </div>
  </div>
</DocumentFragment>
`;
